app/extras/queries/visible_discussions.rb

```diff
@@ -117,7 +116,7 @@ def self.apply_privacy_sql(user: nil, group_ids: [], relation: nil)
                     (discussions.group_id IN (:user_group_ids)) OR
                     (groups.parent_members_can_see_discussions = TRUE AND groups.parent_id IN (:user_group_ids)))',
                    group_ids: group_ids,
-                   user_group_ids: user_group_ids)
+                   user_group_ids: (user || LoggedOutUser.new).group_ids)
```

Comment:
(Because beta calls this and I don't trust that it's always passing in a `current_user_or_visitor` value.)